File: backend-java/src/main/java/com/taru/taskmanager/models/UserProjects.java
package com.taru.taskmanager.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "user_projects")
public class UserProjects {

    @EmbeddedId
    private UserProjectsId id;
    @ManyToOne
    @MapsId("userId")
    @JoinColumn(name = "user_id")
    private User user;
    @ManyToOne
    @MapsId("projectId")
    @JoinColumn(name = "project_id")
    private Project project;
}

File: backend-csharp/TaskManager.Notification/Entities/UserProjects.cs
using System;
using System.Collections.Generic;

namespace TaskManager.Notification.Entities;

public partial class UserProjects
{
    public int ProjectId { get; set; }

    public int UserId { get; set; }

    public virtual Projects Project { get; set; } = null!;

    public virtual Users User { get; set; } = null!;
}
